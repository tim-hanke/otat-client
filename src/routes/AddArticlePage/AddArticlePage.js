import React, { Component } from "react";
import AddArticleForm from "../../components/AddArticleForm/AddArticleForm";
import { Section } from "../../components/Utils/Utils";

export default class AddArticlePage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  };

  handleAddSuccess = () => {
    const { location, history } = this.props;
    const destination = (location.state || {}).from || "/articles";
    history.push(destination);
  };

  render() {
    return (
      <Section className="AddArticlePage">
        <h2>Add Article</h2>
        <AddArticleForm onAddSuccess={this.handleAddSuccess} />
      </Section>
    );
  }
}
