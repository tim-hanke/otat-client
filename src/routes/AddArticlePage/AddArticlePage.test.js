import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import AddArticlePage from "./AddArticlePage";

describe("AddArticlePage component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <AddArticlePage />
      </MemoryRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <AddArticlePage />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
