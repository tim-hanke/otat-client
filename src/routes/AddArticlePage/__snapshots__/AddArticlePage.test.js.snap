// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddArticlePage component renders the UI as expected 1`] = `
<section
  className="Section AddArticlePage"
>
  <h2>
    Add Article
  </h2>
  <form
    className="AddArticleForm"
    onSubmit={[Function]}
  >
    <div
      className="url"
    >
      <label
        htmlFor="AddArticleForm__url"
      >
        URL
      </label>
      <input
        className="Input "
        id="AddArticleForm__url"
        name="url"
        required={true}
      />
    </div>
    <button
      className="Button "
      type="submit"
    >
      Submit URL
    </button>
  </form>
</section>
`;
